src/NUnitFramework/framework/Internal/RuntimeFramework.cs

```diff
@@ -352,13 +385,47 @@ public bool Supports(RuntimeFramework target)
             if (!VersionsMatch(ClrVersion, target.ClrVersion))
                 return false;
 
-            return FrameworkVersion.Major >= target.FrameworkVersion.Major && FrameworkVersion.Minor >= target.FrameworkVersion.Minor;
+            if (FrameworkVersion.Major > target.FrameworkVersion.Major)
+                return true;
+            return FrameworkVersion.Major == target.FrameworkVersion.Major && FrameworkVersion.Minor >= target.FrameworkVersion.Minor;
         }
 
         #endregion
 
         #region Helper Methods
 
+        private static bool IsNetCore()
+        {
+#if NETSTANDARD2_0
+            //Mono versions will throw a TypeLoadException when attempting to run the internal method, so we wrap it in a try/catch 
+            // block to stop any inlining in release builds and check whether the type exists
+            Type runtimeInfoType = Type.GetType("System.Runtime.InteropServices.RuntimeInformation,System.Runtime.InteropServices.RuntimeInformation", false);
+            if (runtimeInfoType != null)
+            {
+                try
+                {
+                    return IsNetCore_Internal();
+                }
+                catch { }
```

Comment:
The usual method to prevent inlining is `[MethodImpl(MethodImplOptions.NoInlining)]`. Would we be able to do that instead?

Reply:
That would help - I'll make that change.

Reply:
Does this change mean the try and catch should also be removed, and comments updated? Or am I missing the point here? This isn't something I'm familiar with! 😅

Reply:
Yes, I would expect that we would remove it. @MatthewBeardmore Was the `try` actually used to catch an exception or just for the side effects on inlining with the current JIT?

Reply:
From my comment below:

For the try/catch addition, it'll get run under older versions of Mono, as when it tries to compile the IsNetCore_Internal method, it will throw an exception as it can't find the System.Runtime.InteropServices assembly, which will be caught in that try/catch.

Reply:
I'd prefer only catching exceptions we specifically know about, like `TypeLoadException`.